python/cuml/metrics/pairwise_distances.pyx

```diff
@@ -136,6 +137,124 @@ def _determine_metric(metric_str, is_sparse=False):
+
+    Compute the euclidean distance between each pair of samples in X and Y,
+    where Y=X is assumed if Y=None. When calculating the distance between a
+    pair of samples, this formulation ignores feature coordinates with a
+    missing value in either sample and scales up the weight of the remaining
+    coordinates:
+
+        dist(x,y) = sqrt(weight * sq. distance from present coordinates)
+        where,
+        weight = Total # of coordinates / # of present coordinates
+
+    For example, the distance between ``[3, na, na, 6]`` and ``[1, na, 4, 5]``
+    is:
+
+        .. math::
+            \\sqrt{\\frac{4}{2}((3-1)^2 + (6-5)^2)}
+
+    If all the coordinates are missing or if there are no common present
+    coordinates then NaN is returned for that pair.
+
+    Read more in the :ref:`User Guide <metrics>`.
+
+    Parameters
+    ----------
+    X : Dense matrix of shape (n_samples_X, n_features)
+        Acceptable formats: cuDF DataFrame, Pandas DataFrame, NumPy ndarray, 
+        cuda array interface compliant array like CuPy.
+
+    Y : Dense matrix of shape (n_samples_Y, n_features), default=None
+        Acceptable formats: cuDF DataFrame, Pandas DataFrame, NumPy ndarray, 
+        cuda array interface compliant array like CuPy.
+
+    squared : bool, default=False
+        Return squared Euclidean distances.
+
+    missing_values : np.nan or int, default=np.nan
+        Representation of missing value.
+
+    Returns
+    -------
+    distances : ndarray of shape (n_samples_X, n_samples_Y)
+        Returns the distances between the row vectors of `X`
+        and the row vectors of `Y`.
+    """
+
+    if isinstance(X, cudf.DataFrame) or isinstance(X, pd.DataFrame):
+        if (X.isnull().any()).any():
+            X.fillna(0, inplace=True)
+
+    if isinstance(Y, cudf.DataFrame) or isinstance(Y, pd.DataFrame):
+        if (Y.isnull().any()).any():
+            Y.fillna(0, inplace=True)
+
+    
+    X_m, n_samples_x, n_features_x, dtype_x = \
+        input_to_cuml_array(X, order="K", check_dtype=[np.float32, np.float64])
+    
+    if Y is None:
+        Y = X_m
+
+    Y_m, n_samples_y, n_features_y, dtype_y = \
+        input_to_cuml_array(Y, order=X_m.order, convert_to_dtype=dtype_x, check_dtype=[dtype_x])
+    
+    X_m = cp.asarray(X_m)
+    Y_m = cp.asarray(Y_m)
+
+    # Get missing mask for X
+    missing_X = _get_mask(X_m, missing_values)
+
+    # Get missing mask for Y
+    missing_Y = missing_X if Y is X else _get_mask(Y_m, missing_values)
+
+    # set missing values to zero
+    X_m[missing_X] = 0
+    Y_m[missing_Y] = 0
+
+    # Adjust distances for sqaured
+    if X_m.shape == Y_m.shape:
+        if (X_m == Y_m).all():
+            distances = cp.array(pairwise_distances(X_m, metric="sqeuclidean"))
+        else:
+            distances = cp.array(pairwise_distances(X_m, Y_m, metric="sqeuclidean"))
+    else:
+        distances = cp.array(pairwise_distances(X_m, Y_m, metric="sqeuclidean"))
```

Comment:
Note: cp.array makes a copy by default. If not needed, should try `asarray` or set `copy=False` (prefer the former)